### ISSUE TITLE
Does core need providers?

### ISSUE THREAD

**Issue by knownasilya** [Strider-CD/core#11]
I'm just wondering if Core really needs providers e.g. Github, Bitbucket, etc, since the work will be done by the drones, core only handles the webhooks to trigger drones.

What would be the benefit of providers on core?

**Comment by microadam** [comment]
"core only handles the webhooks to trigger drones"

Without providers, how can the webhooks be setup and configured? :)

**Comment by knownasilya** [comment]
Ok what I meant by "no providers" is no provider specific APIs apart from webhooks and auth. Should have been more clear.

Really we need to be able to trigger by provider, and convert the providers trigger data to something structured/generic. Setting up webhooks doesn't even have to be automatic right away, since even now we have issues with it, and manual setup isn't so bad (only once per project).

There doesn't need to be any cloning or getting a file (strider.json), or handling ssh keys..

**Comment by phiros** [comment]
I agree that we don't really need providers in the classical sense. If someone needs a functionality where you look into the repo in order determine what to do with the job before sending it to a drone we can implement this a plugin.

BTW: I plan to implement such a plugin very soon after having the basic functionality of core + drone implemented. The use case would be:
- having matrix builds (similar to http://docs.travis-ci.com/user/customizing-the-build/#Build-Matrix)
- distributing jobs to drones which have access or run on specialized hardware (e.g. drone A has access to a piece of hardware Q thus all jobs needing the piece of hardware Q are sent to drone A; this ties in quite nicely with a matrix build concept)
